src/dmd/dmangle.d

```diff
@@ -584,7 +584,8 @@ public:
             assert(slice.length);
             foreach (const char c; slice)
             {
-                assert(c == '_' || c == '@' || c == '?' || c == '$' || isalnum(c) || c & 0x80);
+                assert(c.isValidMangling, "The mangled name '" ~ slice ~ "' " ~
+                    "contains an invalid character: " ~ c);
```

Comment:
Just leave it as:
```
assert(c.isValidMangling);
```
Anyone who cannot look at the file/line of the assert in the source code is not going to be helped by adding a message. Asserts in the compiler need to be utilitarian, so there is no incentive to leave them out of the release build.